mono/metadata/threads.c

```diff
@@ -4529,39 +4529,30 @@ mono_thread_execute_interruption (void)
 
 	/* If there's a pending exception and an AbortRequested, the pending exception takes precedence */
 	if (sys_thread->pending_exception) {
-		MonoException *exc;
-
 		exc = sys_thread->pending_exception;
 		sys_thread->pending_exception = NULL;
-
-		UNLOCK_THREAD (thread);
-		return exc;
-	} else if (thread->state & (ThreadState_AbortRequested)) {
-		UNLOCK_THREAD (thread);
-		g_assert (sys_thread->pending_exception == NULL);
```

Comment:
We lost this assert.

Reply:
On purpose. The code read to me as:
lock
if (foo)
   ..
else if (bar)
   unlock
   assert(!foo)

and I ASSUME foo can only change under lock, and the assert was some attempt to assert the race wasn't lost, while continuing to be just as racy. Given that the if is side-effect free, even if I don't fully understand, the assert still looks silly.